src/main/java/com/coreos/jetcd/lease/KeepAlive.java

```diff
@@ -1,65 +1,72 @@
 package com.coreos.jetcd.lease;
 
-import com.coreos.jetcd.Lease.LeaseHandler;
-import com.coreos.jetcd.api.LeaseGrantResponse;
+import com.coreos.jetcd.LeaseImpl.KeepAliveListenerImpl;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * The KeepAlive hold the keepAlive information for lease.
  */
 public class KeepAlive {
 
-  private final long leaseID;
-
-  private long ttl;
-
-  private LeaseGrantResponse leaseGrantResponse;
-
   private long deadLine;
 
   private long nextKeepAlive;
 
-  private LeaseHandler leaseHandler;
+  private List<KeepAliveListenerImpl> keepAliveListeners = new ArrayList<>();
 
-  public KeepAlive(long leaseID) {
-    this(leaseID, null);
-  }
+  private DoneHandler doneHandler;
 
-  public KeepAlive(long leaseID, LeaseHandler leaseHandler) {
-    this.leaseID = leaseID;
-    this.leaseHandler = leaseHandler;
+  @FunctionalInterface
+  public interface DoneHandler {
```

Comment:
it is not async. I need a way tell LeaseImpl that this keepalive is done where all its listeners are closed by user. Then this keepalive needed be deleted from LeaseImpl's KeepAlives map. The handler is defined in LeaseImpl and all it does is to delete the keepalive.
```java
 @Override
  public synchronized KeepAliveListener keepAlive(long leaseId) {
...
    KeepAlive keepAlive = this.keepAlives.computeIfAbsent(leaseId, (key) -> {
      // the () -> this.removeKeepAlive(leaseId) is handler's definition
      KeepAlive ka = new KeepAlive(() -> this.removeKeepAlive(leaseId)); 
      long now = System.currentTimeMillis();
      ka.setDeadLine(now + FIRST_KEEPALIVE_TIMEOUT_MS);
      ka.setNextKeepAlive(now);
      return ka;
    });
...
```